system/middleware.go

```diff
@@ -98,6 +150,11 @@ func strInSlice(strs []string, str string) int {
 
 func (application *Application) ApplyCsrfProtection(c *web.C, h http.Handler) http.Handler {
 	fn := func(w http.ResponseWriter, r *http.Request) {
+		// disable CSRF for API requests
+		if c.Env["IsAPI"].(bool) {
```

Comment:
Well I added a nil check but again it should be set.  I do agree this could be cleaned up a lot but again I am mostly following the pattern the app has always used which is to execute the middleware in sequence and have the environment properly set up.  Not an unreasonable assumption to make in my opinion.